packages/cloudflare/src/cli/args.ts

```diff
@@ -2,71 +2,61 @@ import { mkdirSync, type Stats, statSync } from "node:fs";
 import { resolve } from "node:path";
 import { parseArgs } from "node:util";
 
-import type { CacheBindingMode } from "./build/utils/index.js";
-import { isCacheBindingMode } from "./build/utils/index.js";
+import { isWranglerTarget, WranglerTarget } from "./utils/run-wrangler.js";
 
-export function getArgs(): {
-  skipNextBuild: boolean;
-  skipWranglerConfigCheck: boolean;
-  outputDir?: string;
-  minify: boolean;
-  populateCache?: { mode: CacheBindingMode; onlyPopulateWithoutBuilding: boolean };
-} {
-  const { skipBuild, skipWranglerConfigCheck, output, noMinify, populateCache, onlyPopulateCache } =
-    parseArgs({
-      options: {
-        skipBuild: {
-          type: "boolean",
-          short: "s",
-          default: false,
-        },
-        output: {
-          type: "string",
-          short: "o",
-        },
-        noMinify: {
-          type: "boolean",
-          default: false,
-        },
-        skipWranglerConfigCheck: {
-          type: "boolean",
-          default: false,
-        },
-        populateCache: {
-          type: "string",
-        },
-        onlyPopulateCache: {
-          type: "boolean",
-          default: false,
-        },
-      },
-      allowPositionals: false,
-    }).values;
+export type Arguments = (
+  | {
+      command: "build";
+      skipNextBuild: boolean;
+      skipWranglerConfigCheck: boolean;
+      minify: boolean;
+    }
+  | { command: "preview" | "deploy"; passthroughArgs: string[] }
+  | { command: "populateCache"; target: WranglerTarget }
+) & { outputDir?: string };
 
-  const outputDir = output ? resolve(output) : undefined;
+export function getArgs(): Arguments {
+  const { positionals, values } = parseArgs({
+    options: {
+      skipBuild: { type: "boolean", short: "s", default: false },
+      output: { type: "string", short: "o" },
+      noMinify: { type: "boolean", default: false },
```

Comment:
Q (not directly related to this PR): aws uses `OPEN_NEXT_DEBUG` to minify, should we do the same and drop this arg?

Reply:
yep, would make sense